"The infrastructure contains errors" message while no error detected in infra editor

### What happened?

I edited an infrastructure, fixed all the errors :
![image](https://github.com/osrd-project/osrd/assets/86296016/37eb40ed-7400-437e-9399-31b66212ca56)
![image](https://github.com/osrd-project/osrd/assets/86296016/84f7f338-3958-4f28-a4c7-92c7ab3181ef)


But still when trying to use it in a scenario it says that my infrastructure has errors:
![image](https://github.com/osrd-project/osrd/assets/86296016/d38e69b9-cda6-4944-ba92-a55ff9fe27e9)


### What did you expect to happen?

If the infrastructure editor says my infrastructure has no errors i should be able to use it.

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to my scenario Alex>Reproduction de bug>Train

### What operating system, browser and environment are you using?

- **Browser**: Chrome
- **OS**: W11
- **Env**: dev dgexsol


### OSRD version (top right corner `Account` button > `Informations`)

faae221

Core stack trace:

```
fr.sncf.osrd.reporting.exceptions.OSRDError: hard error while loading new infra
	at fr.sncf.osrd.reporting.exceptions.OSRDError.newInfraLoadingError(OSRDError.java:141)
	at fr.sncf.osrd.api.InfraManager.downloadInfra(InfraManager.java:181)
	at fr.sncf.osrd.api.InfraManager.load(InfraManager.java:207)
	at fr.sncf.osrd.api.InfraLoadEndpoint.act(InfraLoadEndpoint.java:37)
	at org.takes.facets.fork.FkRegex.lambda$new$0(FkRegex.java:153)
	at org.takes.facets.fork.FkRegex.route(FkRegex.java:217)
	at org.takes.facets.fork.FkChain.route(FkChain.java:72)
	at org.takes.facets.fork.TkFork.act(TkFork.java:98)
	at org.takes.facets.fallback.TkFallback.route(TkFallback.java:84)
	at org.takes.facets.fallback.TkFallback.lambda$new$0(TkFallback.java:66)
	at org.takes.tk.TkWrap.act(TkWrap.java:58)
	at org.takes.tk.TkSlf4j.act(TkSlf4j.java:110)
	at org.takes.http.BkBasic.print(BkBasic.java:123)
	at org.takes.http.BkBasic.accept(BkBasic.java:99)
	at org.takes.http.BkSafe.lambda$new$0(BkSafe.java:46)
	at org.takes.http.BkWrap.accept(BkWrap.java:51)
	at org.takes.http.BkParallel.lambda$new$0(BkParallel.java:81)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.NullPointerException: null
	at fr.sncf.osrd.sim_infra_adapter.RawInfraAdapterKt$adaptRawInfra$2.invoke(RawInfraAdapter.kt:148)
	at fr.sncf.osrd.sim_infra_adapter.RawInfraAdapterKt$adaptRawInfra$2.invoke(RawInfraAdapter.kt:138)
	at fr.sncf.osrd.sim_infra.impl.RawInfraBuilderImpl.movableElement-IvdR598(RawInfraBuilder.kt:305)
	at fr.sncf.osrd.sim_infra_adapter.RawInfraAdapterKt.adaptRawInfra(RawInfraAdapter.kt:138)
	at fr.sncf.osrd.api.InfraManager.downloadInfra(InfraManager.java:158)
fr.sncf.osrd.reporting.exceptions.OSRDError: hard error while loading new infra
	... 18 common frames omitted
	at fr.sncf.osrd.reporting.exceptions.OSRDError.newInfraLoadingError(OSRDError.java:141)
	at fr.sncf.osrd.api.InfraManager.downloadInfra(InfraManager.java:181)
	at fr.sncf.osrd.api.InfraManager.load(InfraManager.java:207)
	at fr.sncf.osrd.api.InfraLoadEndpoint.act(InfraLoadEndpoint.java:37)
	at org.takes.facets.fork.FkRegex.lambda$new$0(FkRegex.java:153)
	at org.takes.facets.fork.FkRegex.route(FkRegex.java:217)
	at org.takes.facets.fork.FkChain.route(FkChain.java:72)
	at org.takes.facets.fork.TkFork.act(TkFork.java:98)
	at org.takes.facets.fallback.TkFallback.route(TkFallback.java:84)
	at org.takes.facets.fallback.TkFallback.lambda$new$0(TkFallback.java:66)
	at org.takes.tk.TkWrap.act(TkWrap.java:58)
	at org.takes.tk.TkSlf4j.act(TkSlf4j.java:110)
	at org.takes.http.BkBasic.print(BkBasic.java:123)
	at org.takes.http.BkBasic.accept(BkBasic.java:99)
	at org.takes.http.BkSafe.lambda$new$0(BkSafe.java:46)
	at org.takes.http.BkWrap.accept(BkWrap.java:51)
	at org.takes.http.BkParallel.lambda$new$0(BkParallel.java:81)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.NullPointerException
	at fr.sncf.osrd.sim_infra_adapter.RawInfraAdapterKt$adaptRawInfra$2.invoke(RawInfraAdapter.kt:148)
	at fr.sncf.osrd.sim_infra_adapter.RawInfraAdapterKt$adaptRawInfra$2.invoke(RawInfraAdapter.kt:138)
	at fr.sncf.osrd.sim_infra.impl.RawInfraBuilderImpl.movableElement-IvdR598(RawInfraBuilder.kt:305)
	at fr.sncf.osrd.sim_infra_adapter.RawInfraAdapterKt.adaptRawInfra(RawInfraAdapter.kt:138)
	at fr.sncf.osrd.api.InfraManager.downloadInfra(InfraManager.java:158)
```

--- a/editoast/openapi.yaml
+++ b/editoast/openapi.yaml
@@ -761,6 +761,7 @@ components:
       - overlapping_catenaries
       - unknown_port_name
       - unused_port
+      - node_endpoints_not_unique
       type: string
     InternalError:
       properties:

--- a/editoast/openapi_legacy.yaml
+++ b/editoast/openapi_legacy.yaml
@@ -1021,6 +1021,7 @@ components:
         - overlapping_catenaries
         - unknown_port_name
         - unused_port
+        - node_endpoints_not_unique
 
     InfraError:
       type: object

--- a/editoast/src/generated_data/error/switches.rs
+++ b/editoast/src/generated_data/error/switches.rs
@@ -1,14 +1,15 @@
 use crate::infra_cache::{Graph, InfraCache, ObjectCache};
-use crate::schema::{InfraError, ObjectRef, ObjectType, TrackEndpoint};
+use crate::schema::{InfraError, OSRDIdentified, ObjectRef, ObjectType, TrackEndpoint};
 use std::collections::{HashMap, HashSet};
 
 use super::ObjectErrorGenerator;
 
-pub const OBJECT_GENERATORS: [ObjectErrorGenerator<Context>; 4] = [
+pub const OBJECT_GENERATORS: [ObjectErrorGenerator<Context>; 5] = [
     ObjectErrorGenerator::new(1, check_invalid_ref_ports),
     ObjectErrorGenerator::new(1, check_invalid_ref_switch_type),
     ObjectErrorGenerator::new(2, check_match_ports_type),
-    ObjectErrorGenerator::new_ctx(3, check_overlapping),
+    ObjectErrorGenerator::new(3, check_endpoints_unicity),
+    ObjectErrorGenerator::new_ctx(4, check_overlapping),
 ];
 
 /// Context for the switch error generators
@@ -86,6 +87,17 @@ pub fn check_match_ports_type(
     }
 }
 
+/// Check if node track endpoints are unique
+pub fn check_endpoints_unicity(switch: &ObjectCache, _: &InfraCache, _: &Graph) -> Vec<InfraError> {
+    let switch = switch.unwrap_switch();
+    let endpoints: HashSet<_> = switch.ports.values().collect();
+    if endpoints.len() != switch.ports.len() {
+        vec![InfraError::new_node_endpoint_not_unique(switch.get_id())]
+    } else {
+        vec![]
+    }
+}
+
 /// Check if the switch ports are not already used by another switch
 fn check_overlapping(
     switch: &ObjectCache,
@@ -114,11 +126,11 @@ fn check_overlapping(
 
 #[cfg(test)]
 mod tests {
-    use crate::generated_data::error::switches::Context;
+    use crate::generated_data::error::switches::{check_endpoints_unicity, Context};
     use crate::infra_cache::tests::{
         create_small_infra_cache, create_switch_cache_point, create_track_endpoint,
     };
-    use crate::schema::{Endpoint, ObjectRef, ObjectType};
+    use crate::schema::{Endpoint, OSRDIdentified, ObjectRef, ObjectType};
 
     use super::check_invalid_ref_ports;
     use super::check_invalid_ref_switch_type;
@@ -164,6 +176,25 @@ mod tests {
         assert_eq!(1, errors.len());
         let obj_ref = ObjectRef::new(ObjectType::SwitchType, "non_existing_switch_type");
         let infra_error = InfraError::new_invalid_reference(&switch, "switch_type", obj_ref);
+        assert_eq!(infra_error, errors[0]);
+    }
+
+    #[test]
+    fn not_unique_endpoints() {
+        let mut infra_cache = create_small_infra_cache();
+        // Ports A and B1 map the same endpoint
+        let switch = create_switch_cache_point(
+            "SW_error".into(),
+            ("A", create_track_endpoint(Endpoint::End, "B")),
+            ("B1", create_track_endpoint(Endpoint::End, "B")),
+            ("B2", create_track_endpoint(Endpoint::Begin, "D")),
+            "point_switch".into(),
+        );
+        infra_cache.add(switch.clone()).unwrap();
+        let errors =
+            check_endpoints_unicity(&switch.clone().into(), &infra_cache, &Default::default());
+        assert_eq!(1, errors.len());
+        let infra_error = InfraError::new_node_endpoint_not_unique(switch.get_id());
         assert_eq!(infra_error, errors[0]);
     }
 

--- a/editoast/src/schema/errors.rs
+++ b/editoast/src/schema/errors.rs
@@ -35,6 +35,7 @@ pub enum InfraErrorType {
     MissingBufferStop {
         endpoint: Endpoint,
     },
+    NodeEndpointsNotUnique,
     ObjectOutOfPath {
         reference: ObjectRef,
     },
@@ -134,6 +135,16 @@ impl InfraError {
             field: Default::default(),
             is_warning: true,
             sub_type: InfraErrorType::MissingRoute,
+        }
+    }
+
+    pub fn new_node_endpoint_not_unique<T: AsRef<str>>(switch_id: &T) -> Self {
+        Self {
+            obj_id: switch_id.as_ref().into(),
+            obj_type: ObjectType::Switch,
+            field: Default::default(),
+            is_warning: false,
+            sub_type: InfraErrorType::NodeEndpointsNotUnique,
         }
     }
 

--- a/front/public/locales/en/translation.json
+++ b/front/public/locales/en/translation.json
@@ -139,6 +139,10 @@
         "unused_port": {
           "name": "Unused branch",
           "description": "Branch « {{port_name}} » of switch type « {{obj_id}} » is nerver used"
+        },
+        "node_endpoints_not_unique": {
+          "name": "Node endpoints not unique",
+          "description": "The « {{obj_id}} » node has a track endpoint used by several ports."
         }
       },
       "list": {

--- a/front/public/locales/fr/translation.json
+++ b/front/public/locales/fr/translation.json
@@ -139,6 +139,10 @@
         "unused_port": {
           "name": "Branche non utilisée",
           "description": "La branche « {{port_name}} » déclarée dans le type d’aiguille/nœud « {{obj_id}} » n’est pas utilisée dans les configurations de ce type"
+        },
+        "node_endpoints_not_unique": {
+          "name": "Extrémité de voie de nœud non unique",
+          "description": "Une extrémité de voie est utilisée par plusieurs ports du nœud « {{obj_id}} »"
         }
       },
       "list": {

--- a/front/src/applications/editor/components/InfraErrors/types.ts
+++ b/front/src/applications/editor/components/InfraErrors/types.ts
@@ -23,6 +23,7 @@ export const infraErrorTypeList: Record<'errors' | 'warnings', Set<InfraErrorTyp
     'object_out_of_path',
     'out_of_range',
     'unknown_port_name',
+    'node_endpoints_not_unique',
   ]),
   warnings: new Set([
     'duplicated_group',
@@ -96,6 +97,9 @@ type InfraErrorUnusedPort = InfraErrorInformation & {
   error_type: 'unused_port';
   port_name: string;
 };
+type InfraErrorNodeEndpointsNotUnique = InfraErrorInformation & {
+  error_type: 'node_endpoints_not_unique';
+};
 
 // Type of an error
 export type InfraError = Omit<InfraErrorApiType, 'informations'> & {
@@ -116,5 +120,6 @@ export type InfraError = Omit<InfraErrorApiType, 'informations'> & {
     | InfraErrorOverlappingSpeedSections
     | InfraErrorOverlappingSwitches
     | InfraErrorOverlappingCatenaries
-    | InfraErrorUnusedPort;
+    | InfraErrorUnusedPort
+    | InfraErrorNodeEndpointsNotUnique;
 };

--- a/front/src/common/api/osrdEditoastApi.ts
+++ b/front/src/common/api/osrdEditoastApi.ts
@@ -1470,7 +1470,8 @@ export type InfraErrorType =
   | 'overlapping_switches'
   | 'overlapping_catenaries'
   | 'unknown_port_name'
-  | 'unused_port';
+  | 'unused_port'
+  | 'node_endpoints_not_unique';
 export type InfraError = {
   geographic?: Geometry | null;
   information: {

--- a/python/osrd_schemas/osrd_schemas/generated.py
+++ b/python/osrd_schemas/osrd_schemas/generated.py
@@ -52,6 +52,10 @@ class InvalidRoute(InfraErrorTrait):
     error_type: Literal["invalid_route"] = Field(default="invalid_route")
 
 
+class NodeEndpointsNotUnique(InfraErrorTrait):
+    error_type: Literal["node_endpoints_not_unique"] = Field(default="node_endpoints_not_unique")
+
+
 class UnknownPortName(InfraErrorTrait):
     error_type: Literal["unknown_port_name"] = Field(default="unknown_port_name")
     port_name: str
@@ -113,6 +117,7 @@ InfraError = Annotated[
         InvalidSwitchPorts,
         MissingRoute,
         MissingBufferStop,
+        NodeEndpointsNotUnique,
         ObjectOutOfPath,
         OddBufferStopLocation,
         OutOfRange,

